graph2tac/loader/predict_server.py

```diff
@@ -343,29 +343,25 @@ def check_alignment(self, msg: CheckAlignmentMessage) -> CheckAlignmentResponse:
             unknown_definitions = unaligned_definitions,
         )
 
-def prediction_loop(predict_server: PredictServer, capnp_socket: socket.socket, record_file: Optional[BinaryIO]):
-    
-    message_generator = capnp_message_generator(capnp_socket, record_file)
+def prediction_loop(predict_server: PredictServer, context: GlobalContextMessage):
```

Comment:
I'm a bit confused by the types here.  This is called below as
```
prediction_loop(predict_server, capnp_message_generator(capnp_socket, record_file))
```
And `capnp_message_generator` returns type
```
Generator[GlobalContextMessage | CheckAlignmentMessage, None | CheckAlignmentResponse, None])
```
I don't even understand how this runs, much alone type checks.  If context is a generator, how does `context.prediction_requests` work?  It obviously does run, so I must be missing something.  How does this work?

Reply:
Nevermind.  I was looking at the wrong branch of coq-tactician-reinforce.  It no longer returns a generator, but a `GlobalContextMessage` which is exactly what we need to make this work.